hoprd/db/api/src/aliases.rs

```diff
@@ -40,53 +44,75 @@ impl HoprdDbMetadataOperations for HoprdDb {
     async fn get_aliases(&self) -> Result<Vec<Alias>> {
         let rows = hoprd_db_entity::aliases::Entity::find().all(&self.metadata).await?;
 
-        let aliases = rows
-            .iter()
-            .map(|row| Alias {
-                peer_id: row.peer_id.clone(),
-                alias: row.alias.clone(),
-            })
-            .collect();
+        let aliases: Vec<Alias> = rows.iter().map(|row| Alias::from(row.to_owned())).collect();
 
         Ok(aliases)
     }
 
     async fn set_aliases(&self, aliases: Vec<Alias>) -> Result<()> {
+        if let Some(me) = self.resolve_alias(ME_AS_ALIAS.to_string()).await.unwrap() {
+            if aliases.iter().any(|entry| entry.peer_id == me) {
+                return Err(crate::errors::DbError::LogicalError(
+                    "own alias cannot be modified".into(),
+                ));
+            }
+        }
+
         let new_aliases = aliases
             .iter()
             .map(|entry| hoprd_db_entity::aliases::ActiveModel {
+                id: Default::default(),
                 peer_id: sea_orm::ActiveValue::Set(entry.peer_id.clone()),
                 alias: sea_orm::ActiveValue::Set(entry.alias.clone()),
-                ..Default::default()
             })
             .collect::<Vec<_>>();
 
         let _ = hoprd_db_entity::aliases::Entity::insert_many(new_aliases)
+            .on_conflict(
+                OnConflict::columns([Column::PeerId])
+                    .update_column(Column::PeerId)
```

Comment:
Normally, if you conflict on column `A`, you'd update the other column `B`.
This would me it's actually a no-op (= you're updating the column on which you're conflicting).